<launch>
  <node pkg="bw_auto_dock" type="bw_auto_dock" name="dock_driver">
    <!-- 充电部位到base_link位置的距离，base_link一般位于两个主动轮轴中间-->
    <param name="back_distance" value="0.40"/>
    <!-- 自动充电对准过程中的最大直线速度-->
    <param name="max_linearspeed" value="0.20"/>
    <!-- 自动充电对准过程中的最大旋转速度-->
    <param name="max_rotspeed" value="1.0"/>
    <!-- 自动充电直线运动对准阶段中的pid参数-->
    <param name="back_dock_kp" value="0.20"/>
    <param name="back_dock_ki" value="0.04"/>
    <param name="back_dock_kd" value="0.0"/>
    <!-- 自动充电设备串口号-->
    <param name="port" value="/dev/ttyUSB004"/>
    <!-- 里程计所在坐标系名字-->
    <param name="odom_frame_id" value="map"/>
    <!-- 里程计话题名字-->
    <remap from="/odom" to="/bWmono/Odom" />
    <!-- 保存有充电桩位置的文件名,需要是绝对路径，xiaoqiang改成自己的用户名 -->
    <param name="station_filename" value="/home/xiaoqiang/slamdb/dock_station.txt"/>
    <param name="grid_length" value="4.0"/>
    <!-- 碰撞触发距离，单位mm -->
    <param name="crash_distance" value="120"/>
  </node>
  <!-- 设为０关闭底盘红外，设为１启动底盘红外-->
  <!-- <node pkg="nav_test" type="barDetectChanger.py" name="dock_barDetectChanger_node">
    <param name="barDetectFlag" value="0"/>
  </node> -->
</launch>
